### ISSUE TITLE
Low precision of division by scalar in CPU mode.

### ISSUE THREAD

**Issue by mazeltovlee** [pytorch/pytorch#88965]
### 🐛 Describe the bug

When I am using PyTorch, I notice that the computation precision and output precision of the division operator follows the precision of non-scalar operands. This design may decreases the precision when I am doing division by scalar. Previous PR https://github.com/pytorch/pytorch/commit/324c18fcad579b1afa63ae45528bf598ba8ec4ca gives a nice fix for it. In particular, this pr increases the half precision inferred by `scalar_t` to float precision `accscalar_t` so the computation of division is performed in a high precision mode and no precision of denominator is lost. However, the issue is only fixed at the CUDA level while persists at the CPU level. Here is a code snippet to reproduce this issue:

```
import torch
x_cpu = torch.tensor([3388.]).half().to("cpu")
x_gpu = torch.tensor([3388.]).half().to("cuda:0")
scale = 524288.0
print(x_cpu.div(scale), x_gpu.div(scale))

output:
tensor([0.], dtype=torch.float16) tensor([0.0065], device='cuda:0', dtype=torch.float16)
```

I would be much appreciate if you can help fix this issue and make the computation results between CPU and GPU consistent.

### Versions

Collecting environment information...
PyTorch version: 1.12.1+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.22.6
Libc version: glibc-2.26

Python version: 3.7.15 (default, Oct 12 2022, 19:14:55)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.10.133+-x86_64-with-Ubuntu-18.04-bionic
Is CUDA available: True
CUDA runtime version: 11.2.152
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 460.32.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.6
[pip3] torch==1.12.1+cu113
[pip3] torchaudio==0.12.1+cu113
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.13.1
[pip3] torchvision==0.13.1+cu113
[conda] Could not collect

**Comment by ngimel** [comment]
fp16 cpu support generally doesn't adhere to any precision standards, and should not be relied upon.

**Comment by leslie-fang-intel** [comment]
@mazeltovlee Could you try to convert the scalar into a 1-dim tensor? like this:
```
import torch
x_cpu = torch.tensor([3388.]).half().to("cpu")
x_gpu = torch.tensor([3388.]).half().to("cuda:0")
scale = torch.tensor([524288.0])
print(x_cpu.div(scale), x_gpu.div(scale.to("cuda:0")))
```
In this way, CPU and GPU will have same result as:
```
tensor([0.0065]) tensor([0.0065], device='cuda:0')
```

**Comment by mazeltovlee** [comment]
I see.... 
Thanks for your explanation @ngimel and thanks for your solution @leslie-fang-intel. 
I am closing this issue.

**Comment by ngimel** [comment]
Please note that if you convert `scale` to 1d tensor, result is fp32, not fp16 as in original example.
